File: src/main/frontend/proyectoIntegrado/src/app/models/Film.ts
import {Schedule} from "./Schedule";
import {Director} from "./Director";
import {Actor} from "./Actor";
import {Genre} from "./Genre";

export interface Film {
  idFilm: number;
  title: string;
  duration: number;
  releaseDate: Date;
  sinopsis: String;
  score: number;
  photoUrl: string;
  trailer: string;
  directors: Director[];
  actors: Actor[];
  schedules: Schedule[];
  genres: Genre[];
}

File: src/main/java/org/iesbelen/proyecto_integrado/domain/Film.java
package org.iesbelen.proyecto_integrado.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.*;

@Entity
@Table(name = "film")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Film {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_film")
    private long idFilm;
    private String title;
    private int duration;
    private Date releaseDate;
    private String sinopsis;
    private Double score;
    private String photoUrl;
    private String trailer;

    @OneToMany(mappedBy = "film", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Director> directors = new ArrayList<>();

    @OneToMany(mappedBy = "film", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Actor> actors = new ArrayList<>();

    @OneToMany(mappedBy = "film", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Schedule> schedules = new ArrayList<>();

    @OneToMany(mappedBy = "film", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Genre> genres = new ArrayList<>();
}
